### PR TITLE
Add Firmata integration docs

### PR DESCRIPTION
**Pull request by DaAwesomeP**
## Proposed change

Adds documentation for Firmata component (pull open in core repo).

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/35591
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/1633
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by DaAwesomeP** [comment]
I have opened a brands pull and updated the checklist above.

File: source/_integrations/firmata.markdown [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: "0.110"
+ha_iot_class: Local Push
+ha_config_flow: false
+ha_codeowners:
+  - '@DaAwesomeP'
+ha_domain: firmata
+redirect_from:
```
Comment (klaasnicolaas) [inline source/_integrations/firmata.markdown]: This option is no longer needed, you can remove it.

File: source/_integrations/firmata.markdown [hunk 2]
```diff
@@ -0,0 +1,153 @@
+---
+title: Firmata
+description: "Connect Arduino-compatible boards within Home Assistant"
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: "0.110"
```
Comment (klaasnicolaas) [inline source/_integrations/firmata.markdown]:
```suggestion
ha_release: 0.111
```

File: source/_integrations/firmata.markdown [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ha_iot_class: Local Push
+ha_config_flow: false
```
Comment (klaasnicolaas) [inline source/_integrations/firmata.markdown]: If config flow isn't support for now you can remove this line

File: source/_integrations/firmata.markdown [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+- [Binary Sensor](#binary_sensor)
+- [Switch](#switches)
+
+## Configuration
+
+You will need to upload a [Firmata firmware](https://github.com/firmata/) to your board. Please upload any `StandardFirmata` sketch to your board; please refer to the [Arduino documentation](https://www.arduino.cc/en/Main/Howto) for further information for uploading sketches. Firmata is available in the Arduino Library Manager in the Arduino IDE. You may also use a [Firmata Express](https://github.com/MrYsLab/FirmataExpress).
+
+To integrate an Arduino boards with Home Assistant, add the following section to your `configuration.yaml` file:
+
```
Comment (klaasnicolaas) [inline source/_integrations/firmata.markdown]:
```suggestion
```

Please remove the extra blank lines (except 1 :smile: )

File: source/_integrations/firmata.markdown [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+
+<div class='note warning'>
+Firmata does not store last state accross power cycles. This means that with every powerup the pins are set to off/low and after every initialization they will be set to the default that you have configured in YAML.
+</div>
+
+<div class='note warning'>
+Firmata may cycle pins HIGH/LOW during initialization. If your pins should not be accidentally cycled, you may need to modify your Firmtata sketch or if you are using a relay board use an output pin as VCC. See this issue for details: <a href="https://github.com/firmata/arduino/issues/166" target="_blank">Make initial pin value configurable</a>
```
Comment (klaasnicolaas) [inline source/_integrations/firmata.markdown]: It's also possible to use the markdown style for the link to Github: [See our docs](https://developers.home-assistant.io/docs/documenting/create-page#html)

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_integrations/firmata.markdown (hunk 5 above)
Comment (DaAwesomeP) [reply source/_integrations/firmata.markdown]: I tried but it doesn't seem to work inside the `<div>` unless I typoed it. I will try again.

Comment (DaAwesomeP) [reply source/_integrations/firmata.markdown]: Oh I see! I need a newline!

**Comment by DaAwesomeP** [comment]
I have pushed changes fixing all of the issues. Thanks!

File: source/_integrations/firmata.markdown [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+There is currently support for the following device types within Home Assistant:
+
+- [Binary Sensor](#binary_sensor)
+- [Switch](#switches)
+
+## Configuration
+
+You will need to upload a [Firmata firmware](https://github.com/firmata/) to your board. Please upload any `StandardFirmata` sketch to your board; please refer to the [Arduino documentation](https://www.arduino.cc/en/Main/Howto) for further information for uploading sketches. Firmata is available in the Arduino Library Manager in the Arduino IDE. You may also use a [Firmata Express](https://github.com/MrYsLab/FirmataExpress).
+
+To integrate an Arduino boards with Home Assistant, add the following section to your `configuration.yaml` file:
```
Comment (frenck) [inline source/_integrations/firmata.markdown]:
```suggestion
To integrate Arduino boards with Home Assistant, add the following section to your `configuration.yaml` file:
```

File: source/_integrations/firmata.markdown [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+<div class='note warning'>
+Firmata does not store last state accross power cycles. This means that with every powerup the pins are set to off/low and after every initialization they will be set to the default that you have configured in YAML.
```
Comment (frenck) [inline source/_integrations/firmata.markdown]:
```suggestion
Firmata does not store the last state across power cycles. This means that with every powerup the pins are set to off/low and after every initialization, they will be set to the default that you have configured in YAML.
```

File: source/_integrations/firmata.markdown [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+      type: integer
+    negate:
+      description: Flips the input of the digital pin
+      required: false
+      default: False
+      type: boolean
+{% endconfiguration %}
+
+<div class='note'>
+NOTE: If you double-configure a pin, the integration will fail to configure the second one that it attempts to setup and will log an error.
```
Comment (frenck) [inline source/_integrations/firmata.markdown]: No need to add a note, as it is already displayed as a note.
```suggestion
If you double-configure a pin, the integration will fail to configure the second one that it attempts to set up and will log an error.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @DaAwesomeP.

I've reviewed it and left a couple of comments, could you take a look? Thanks! 👍

**Comment by DaAwesomeP** [comment]
@frenck Fixed! Thanks!

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Comment by DaAwesomeP** [comment]
I have updated `ha_release` to 0.112 since 0.111 was released.

**Comment by DaAwesomeP** [comment]
I have updated ha_release to 0.113 since 0.112 was released.

**Comment by DaAwesomeP** [comment]
:tada: home-assistant/core#35591 has been merged

File: source/_integrations/firmata.markdown [hunk 9]
```diff
@@ -0,0 +1,149 @@
+---
+title: Firmata
+description: "Connect Arduino-compatible boards within Home Assistant"
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: 0.113
```
Comment (klaasnicolaas) [inline source/_integrations/firmata.markdown]:
```suggestion
ha_release: 0.114
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Review by frenck — APPROVED** [review]

**Pull request merged**